### ISSUE TITLE
GFF3Tabix histograms can be glitchy

### ISSUE THREAD

**Issue by cmdcolin** [GMOD/jbrowse#1192]
This was an issue raised during recent dev call. Mentioned that histograms and features can be displayed at the same time. If you have any more information or test data feel free to add @enuggetry

**Comment by cmdcolin** [comment]
Some of these glitches should be fixed after merge

I am seeing one more issue that can occur on initial track load where the Y scale is displayed even when features are displayed

Essentially there is a weird thing where at the start JBrowse views at some sort of wider scale (scale = 0.001 for example) before relocating to the region to display (which might have more zoomed in scale 0.05 for example) and then this causes the YScale to be created initially but seems to be a failure to remove it

See the screenshot
![screenshot-localhost-2018 08 29-11-16-46](https://user-images.githubusercontent.com/6511937/44797518-5fb8ac00-ab7d-11e8-9547-d79e925b94c6.png)

After you move back and forth it does go away but the initial display should ideally not contain it. This code for removing it seemed to not be working and I could only think of a setTimeout solution

```
diff --git a/src/JBrowse/View/Track/_YScaleMixin.js b/src/JBrowse/View/Track/_YScaleMixin.js
index 7f3c9d2b4..92087fce8 100644
--- a/src/JBrowse/View/Track/_YScaleMixin.js
+++ b/src/JBrowse/View/Track/_YScaleMixin.js
@@ -86,15 +86,14 @@ return declare( null, {
      * @private
      */
     removeYScale: function() {
-        if( !this.yscale ) {
-            query( '.ruler', this.div ).orphan();
-            return;
-        }
-        if( this.yscale ) {
-            this.yscale.parentNode.removeChild( this.yscale );
-            delete this.yscale;
-        }
-        delete this.yscale_params;
+        setTimeout(() => {
+            console.log('this',this.yscale)
+            if( this.yscale ) {
+                this.yscale.parentNode.removeChild( this.yscale );
+                delete this.yscale;
+            }
+            delete this.yscale_params;
+        },500)
     },

```

**Comment by cmdcolin** [comment]
Closed after 1.15.3 release, the histogram y scale issue is a tough one to solve at time being with jbrowse 1 codebase
